packages/contracts/test/chugsplash/storage.spec.ts

```diff
@@ -0,0 +1,252 @@
+import { expect } from '../setup'
```

Comment:
This seems like a solid first pass test suite, but I do think that we are going to want more coverage. It seems like this could really benefit from fuzz testing. Also we'll at some point want this codepath audited -- cc @ben-chain @maurelian 

That said, for the time being one thing we can do to mitigate risk is a) only allow simple, non-dynamic types for fields, and b) run post checks after each deployment on all fields to make sure they are set correctly. Does that line up with your thinking @smartcontracts ?

Reply:
Good call out, I just started a document for scoping/planning the upcoming audit in June. Filed under [Engineering Notes](https://www.notion.so/optimismpbc/Engineering-Notes-15bd6b2146164cc3af2f6497a31796b9).

Reply:
I thought about this more, I think a better test suite would be to add a `setStorage` function that can change storage slots directly, then make those variables public, then retrieve the values from the contract & assert that their value matches the expected value.